onnxruntime/core/providers/cuda/tensor/gather_elements.cc

```diff
@@ -9,115 +9,94 @@
 namespace onnxruntime {
 namespace cuda {
 
+// Ideally both input and indices can support strided tensor, for training case, the indices is the input for both
+// GatherElements and GatherElementsGrad, indices supporting strided tensor is more useful for saving memory.
+// So we only mark indices as MayStridedInput for now. Will do this for input once needed.
+#ifdef ENABLE_TRAINING
+#define CREATE_GATHER_ELEMENTS_KERNEL_DEF (*KernelDefBuilder::Create()).MayStridedInput(1)
```

Comment:
Same comment as above - Change to more intuitive name like SupportStridedInput or StridedInputSupported